File: lims-web-app/src/services/departmentService.js
import axios from "axios";
import { Urls } from "./apiURLS";

export const departmentsGetAll = async (pageNumber, pageSize, startsWith = '', createdBy) => {
  try {
    const response = await axios.post(
      `${Urls.MASTERS}department/get-all?pageNumber=${pageNumber}&pageSize=${pageSize}&sortBy=name&startsWith=${startsWith}`, {}, {
      headers: {
        createdBy
      }
    });
    return response.data;
  } catch (error) {
    console.error('Error fetching departments:', error);
    return error;
  }
};

export const addDepartment = async (department, createdBy) => {
  try {
    const response = await axios.post(`${Urls.MASTERS}department/save`, department, {
      headers: {
        'createdBy': createdBy
      }
    });
    return response.data;
  } catch (error) {
    console.error('Error adding department:', error);
    return error;
  }
};

export const deleteDepartment = async (id) => {
  try {
    const response = await axios.delete(`${Urls.MASTERS}department/${id}`);
    return response?.data;
  } catch (error) {
    console.error('Error deleting department:', error);
    return error;
  }
};

export const updateDepartment = async (id, department, userId) => {
  try {
    const response = await axios.put(`${Urls.MASTERS}department/update/${id}`, department, {
      headers: {
        'userId': userId
      }
    });
    return response.data;
  } catch (error) {
    console.error('Error updating department:', error);
    return error;
  }
};

export const getDepartmentById = async (id) => {
  try {
    const response = await axios.get(`${Urls.MASTERS}department/${id}`);
    return response.data;
  } catch (error) {
    console.error('Error fetching department by id:', error);
    return error;
  }
};

File: back-end-lims/master-data-service/java/com/digiworldexpo/lims/master/service/DepartmentService.java
package com.digiworldexpo.lims.master.service;

import java.util.List;


import java.util.UUID;

import com.digiworldexpo.lims.entities.master.Department;
import com.digiworldexpo.lims.master.model.response.ResponseModel;

public interface DepartmentService {


	ResponseModel<Department> saveDepartment(Department department, UUID createdBy);

	
	ResponseModel<List<Department>> getDepartments(String startsWith, int pageNumber, int pageSize, String sortBy,
			UUID createdBy);


	ResponseModel<Department> getDepartmentById(UUID id);

	ResponseModel<Department> updateDepartment(UUID id, Department updatedDepartment, UUID userId);

	ResponseModel<Department> deleteDepartment(UUID id);


}